server/server.go

```diff
@@ -69,6 +73,7 @@ type Server struct {
 	grTmpClients  map[uint64]*client
 	grRunning     bool
 	grWG          sync.WaitGroup // to wait on various go routines
+	cproto        int64          // number of clients supporting async INFO
```

Comment:
No we can't because as I describe, the server builds the array of clients, release the lock and then goes over the list of clients to pick the one that needs to receive INFO. In order to use `cproto` for the size of the array, I would need to do the checks that I do after the server lock is released inside the server lock, which means grabbing clients' lock inside server lock and later, outside of the server lock to do the tcp sends. It is possible to do so, just let me know. I guess that your only concern was doing the sends inside the server lock, right?